### ISSUE TITLE
Stay in the same group after closing the last tab in that group

### ISSUE THREAD

**Issue by Beexx** [projectdelphai/panorama-tab-groups#108]
Hello,
I've been trying this extension on Firefox for a while, and I find it great. 
One thing that annoys me a bit is when you're closing the last tab of a group (the one on the right), it sometimes switches to another group. I would love to stay in the same group after closing this tab.
I'm guessing it has something to do with Firefox policy after closing tabs: does it open the tab from which the closed tab was launched, for instance?
Thanks,

**Comment by Beexx** [comment]
Edit: this happens not only for the last tab. I've just seen the same behaviour after closing a tab in the middle of a group: it switched to another tab in another group instead of the currently open group.
I was not able to reproduce this, as I don't understand the logic behind it.

**Comment by roeyk** [comment]
Yes, let's pleeeeease get this fixed, it's been driving me absolutely nuts for months now.

**Comment by vaxul** [comment]
I've tried it and can't reproduce it.

Would you both please provide detailed steps to reproduce this behaviour?

**Comment by Beexx** [comment]
Actually, I can't reproduce it myself... At some point I installed the "Tree Style Tab" extension, and I'm using it now instead of Panorama Tab Groups. Maybe some behaviour of Tree Style Tab made it work somehow in Panorama. I've disabled temporarily Tree Style Tab to see if I could reproduce the issue on Panorama, but I wasn't able to reproduce the issue. Maybe roeyk can help?

**Comment by roeyk** [comment]
Alright; so, first, I have probably > 100 tabs spread out across eight tab groups.  If I close the last tab in one group, I get moved to another group instead of the next last open tab in the same group.

**Comment by vaxul** [comment]
@roeyk
Would you precisely explain what do you mean with "last tab".
F.e. 3 tabs open and close the third or 3 tabs open and close all tabs.

**Comment by roeyk** [comment]
So by the "last tab", I mean that, for example, if I have a group with
three tabs in it and close the third one, then I get switched to another
group completely instead of to the remaining tab in the same group.

**Comment by vaxul** [comment]
Thanks for the clarification. I can't reproduce this? Would you please share some more details about your system? OS,  Firefox version and so on. Maybe capture a video?

**Comment by projectdelphai** [comment]
I've experienced this before and I have some hunches on where to look for this. Try opening a couple groups with a couple tabs in each group. Then move a tab from group #1 to group #3 for example and then close that tab in group #3. I think we might be getting thrown back to group #1 because that's where it came from, we had an issue like this before.

The other option which could be it is to create the same scenario but make two consecutive tabs in one group and move the second to group #3, I think you'll jump back to group #1 because the last created tab (or maybe last used?) is in group #1. 

Either way easiest way to fix this  should be as easy as just: when a tab is closed, make the next active tab the one right to the left (or right) of the just closed tab. Make the current active group the current active group (just to verify)

**Comment by mozhin** [comment]
Select After Closing Current addon solves this problem.
